documentation/vnet2vnet-service/design/vnet-to-vnet-service.md

```diff
@@ -0,0 +1,312 @@
+In the outbound flow, the criteria listed below are applied.
+
+- The appliance assumes it is the first appliance to apply the policy. 
+- It applies the outbound ACLs in three stages: **VNIC**, **Subnet** and
+  **VNET**. The stages are processed in order, with the outcome being the most
+  restrictive of the three ACLs combined.
+- After the ACL stages, the appliance performs **LPM routing** based on the
+  **inner dst-ip** and applies the respective action: **encap**, **subsequent**
+  **CA-PA mapping**.
+- Finally, it **updates** the **connection tracking table** for both inbound and
+  outbound.
+
+### Inbound packet processing pipeline
+
+![dash-outbound](../../general/design/images/dash-hld-inbound-packet-processing-pipeline.svg)
+
+<figcaption><i>Figure 3 - Intbound packet processing pipeline</i></figcaption><br/><br/>
+
+- The pipeline sets the direction as RX(**inbound**) based on the **incoming
+  packet's VNI**, if it does not match against any reserved VNI.
+- Using the inner `dst-mac`, **maps to the corresponding ENI**. 
+- In the inbound flow, Routing (LPM) lookup happens based on VNI and SRC PA
+  prefix and maps to VNET. 
+- Using the VNET mapping tables, source PA address is validated against the list
+  of mappings. If the check passes, **decap** action is performed, else dropped. 
+- After LPM, the three ACL stages are processed in order. ACLs can have multiple
+  `src/dst` IP ranges or port ranges as match criteria.
+
+> [!NOTE] CA-PA mapping table are used for both encap and decap processing.
+
+## VM to VM communication in VNET example
+
+The configuration example shown below defines the values for VNET, ENI, routing
+types, match/action tables, and routing tables.
+
+
+### Configuration example
+
+``` 
+
+/* Define Vnet1 */
+
+DASH_VNET:Vnet1: {
+    "vni": 45654,
+    "guid": "559c6ce8-26ab-4193-b946-ccc6e8f930b2"
+}
+
+/* Define ENI */
+DASH_ENI:F4939FEFC47E : { 
+    "eni_id": "497f23d7-f0ac-4c99-a98f-59b470e8c7bd",
+    "mac_address": "F4939FEFC47E",
+    "pa_addr": 25.1.1.1,
+    "admin_state": "enabled",
+    "vnet": "Vnet1"
+}
+
+/* Define routing types */
+
+DASH_ROUTING_TYPE:vnet: [
+    {
+         "name": "action1", 
+         "action_type: "maprouting" 
+    } 
+]
+
+DASH_ROUTING_TYPE:vnet_direct: [
+    {
+         "name": "action1", 
+         "action_type: "maprouting" 
+    }
+]
+
+DASH_ROUTING_TYPE:vnet_encap: [
+    {
+         "name": "action1",
+         "action_type: "staticencap",
+         "encap_type" "vxlan"
+    }
+]
+
+DASH_ROUTING_TYPE:privatelink: [
+    {
+         "name": "action1",
+         "action_type:"staticencap",
+         "encap_type":"vxlan"
+    },
+    {
+         "name": "action2",
+         "action_type:"staticencap",
+         "encap_type":"nvgre",
+         "vni":100
+    }
+]
+
+/* Define routing tables */
+
+DASH_ROUTE_TABLE:F4939FEFC47E:10.1.0.0/16: {
+    "action_type":"vnet",
+    "vnet":"Vnet1"
+}
+
+DASH_ROUTE_TABLE:F4939FEFC47E:10.1.0.0/24: {
+    "action_type":"vnet",
```

Comment:
@KrisNey-MSFT please fix this as well. change action_type from vnet to vnet_direct

Reply:
Edits made above, please review thank you

Reply:
@KrisNey-MSFT i think you changed the action type for the wrong entry above. It needs to be changed for this one 10.1.0.0./24 since this is the one that has the overlay_ip field.